packages/react-components/react-motion/stories/src/Motion/CustomMotion.stories.tsx

```diff
@@ -0,0 +1,137 @@
+    },
+  };
+});
+
+const RadialBackdropMotion = createPresenceComponent(() => {
+  const keyframes = [
+    {
+      opacity: 0,
+      background: 'radial-gradient(circle at center, rgba(0, 0, 0, 0) 0%, rgba(0, 0, 0, 0.8) 100%)',
+      transform: 'scale(0)',
+    },
+    {
+      opacity: 1,
+      background: 'radial-gradient(circle at center, rgba(0, 0, 0, 0) 0%, rgba(0, 0, 0, 0.8) 100%)',
+      transform: 'scale(1)',
+    },
+  ];
+
+  return {
+    enter: {
+      keyframes: keyframes,
+      easing: motionTokens.curveDecelerateMax,
+      duration: motionTokens.durationGentle,
+    },
+    exit: {
+      keyframes: [...keyframes].reverse(),
+      easing: motionTokens.curveAccelerateMax,
+      duration: motionTokens.durationGentle,
+    },
+  };
+});
+
+export const CustomMotion = () => {
+  const classes = useStyles();
+
+  return (
+    <div className={classes.wrapper}>
+      <Dialog
+        surfaceMotion={{
+          children: (_, props) => <SlideDialogMotion {...props}>{props.children}</SlideDialogMotion>,
+        }}
+      >
+        <DialogTrigger disableButtonEnhancement>
+          <Button>Custom surface motion</Button>
+        </DialogTrigger>
+        <DialogSurface>
+          <DialogBody>
+            <DialogTitle>Custom surface motion</DialogTitle>
+            <DialogContent>
+              If you need to use a custom animation, you can use a render function inside <b>surfaceMotion</b> slot and
+              apply your custom animation built with{' '}
+              <Link href="https://react.fluentui.dev/?path=/docs/motion-apis-createpresencecomponent--docs">
+                createPresenceComponent
+              </Link>
+            </DialogContent>
+            <DialogActions>
+              <DialogTrigger disableButtonEnhancement>
+                <Button appearance="secondary">Close</Button>
+              </DialogTrigger>
+              <Button appearance="primary">Do Something</Button>
+            </DialogActions>
+          </DialogBody>
+        </DialogSurface>
+      </Dialog>
+      <Dialog>
+        <DialogTrigger disableButtonEnhancement>
+          <Button>Custom backdrop</Button>
+        </DialogTrigger>
+        <DialogSurface
+          backdropMotion={{
+            children: (_, props) => <RadialBackdropMotion {...props}>{props.children}</RadialBackdropMotion>,
+          }}
+        >
+          <DialogBody>
+            <DialogTitle>Custom backdrop</DialogTitle>
+            <DialogContent>Customized dimmed background motion</DialogContent>
+            <DialogActions>
+              <DialogTrigger disableButtonEnhancement>
+                <Button appearance="secondary">Close</Button>
+              </DialogTrigger>
+              <Button appearance="primary">Do Something</Button>
+            </DialogActions>
+          </DialogBody>
+        </DialogSurface>
+      </Dialog>
+    </div>
+  );
+};
+
+CustomMotion.parameters = {
+  docs: {
+    description: {
+      story,
```

Comment:
lets inline this, the content is simple enough. with any `import` we create invalid stackblitz demo -> we process and export only source file of story definition